### PR TITLE
Add metric <> tasktype validation logic

### PR DESCRIPTION
**Pull request by kicha0**
This validates that user input requested metric for scorecard is checked against the expected task type.

### CODE REVIEW

File: src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py [hunk 1]
```diff
@@ -283,6 +293,11 @@ def __init__(self, raiinsightdata, config):
             self.pos_label = str(self.data._classes[0])
             self.classes = [self.other_class, self.pos_label]
 
+    def validate_valid_metric_for_task_type(self):
+        for m in self.metrics:
+            if m in metric_task_map and metric_task_map[m].lower() != self.tasktype:
```
Comment (tongyu-microsoft) [inline src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py:298]: Do we need to check for regression metrics?

Comment (tongyu-microsoft) [inline src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py:298]: is this `.lower()` needed here for `metric_task_map[m]`? If yes, do we want to add `.lower()` for `self.tasktype` as well?

**Review by tongyu-microsoft — APPROVED** [review]

File: src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py (hunk 1 above)
Comment (kicha0) [reply src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py:298]: Within this module, we have supplemented some additional metrics mapping for metrics defined within the module, like the following:
metric_task_map["error_rate"] = "classification"

We have not had to supplement any regression metrics and they come from:
from erroranalysis._internal.metrics import metric_to_func, metric_to_task

Comment (kicha0) [reply src/responsibleai/rai_analyse/_score_card/_rai_insight_data.py:298]: self.tasktype is assigned either "regression" or "classification" in the logic of data generation class constructor.

**Pull request merged**